File: easy/987.BinaryNumberwithAlternatingBits/s.py
class Solution:
    """
    @param n: a postive Integer
    @return: if two adjacent bits will always have different values
    """
    def hasAlternatingBits1(self, n):
        # Write your code here
        b = bin(n)
        for i in range(len(b)-1):
            if b[i] == b[i+1]:
                return False
        return True
    def hasAlternatingBits(self, n):
        # Write your code here
        bits = bin(n)
        return all(bits[i] != bits[i+1] for i in xrange(len(bits)-1))

File: easy/987.BinaryNumberwithAlternatingBits/s.go
import (
	"strconv"
)

func hasAlternatingBits(n int) bool {
	// Write your code here
	bits := strconv.FormatInt(int64(n), 2)
	for i := 0; i < len(bits)-1; i++ {
		if bits[i] == bits[i+1] {
			return false
		}
	}
	return true
}